### PR TITLE
Flexible bin unit in timedelta plot

### PR DESCRIPTION
**Pull request by sarakolding**
- [x] I have considered whether this PR needs review, and requested a review if necessary.

Fixes issue #

# Notes for reviewers
Reviewers can skip X, but should pay attention to Y.

### CODE REVIEW

**Comment by sarakolding** [comment]
tried to google, but can't figure out how to solve mypy error 😅

File: src/psycop_model_evaluation/binary/time/timedelta_plots.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def plot_time_from_first_positive_to_event(
     eval_dataset: EvalDataset,
     min_n_in_bin: int = 0,
     bins: Sequence[float] = tuple(range(0, 36, 1)),  # noqa
+    bin_unit: Literal["h", "D", "M", "Q", "Y"] = "M",
```
Comment (MartinBernstorff) [inline src/psycop_model_evaluation/binary/time/timedelta_plots.py:160]: We probably want to extract `Literal["h", "D", "M", "Q", "Y"]` into its own variable, e.g.:

`bin_unit_literals = Literal["h", "D", "M", "Q", "Y"]`

And then use it everywhere, so we don't need to keep them in sync:

```suggestion
    bin_unit: bin_unit_literals = "M",
```

File: src/psycop_model_evaluation/binary/time/timedelta_plots.py [hunk 2]
```diff
@@ -167,6 +168,7 @@ def plot_time_from_first_positive_to_event(
         eval_dataset: EvalDataset object
         min_n_in_bin (int): Minimum number of patients in each bin. If fewer, bin is dropped.
         bins (Sequence[float]): Bins to group by. Defaults to (5, 25, 35, 50, 70).
+        bin_unit (Literal["H", "D", "M", "Q", "Y"], optional): Unit of time to bin by. Defaults to "M".
```
Comment (MartinBernstorff) [inline src/psycop_model_evaluation/binary/time/timedelta_plots.py:171]:
```suggestion
        bin_unit (Literal["h", "D", "M", "Q", "Y"]): Unit of time to bin by. Defaults to "M".
```

Not optional, e.g. doesn't accept `None`.

File: src/psycop_model_evaluation/binary/time/timedelta_plots.py [hunk 3]
```diff
@@ -196,10 +198,12 @@ def plot_time_from_first_positive_to_event(
...
-    df_true_pos["time_from_pred_to_event"] = (
-        df_true_pos["time_from_pred_to_event"] / timedelta(days=1)  # type: ignore
-    ).astype(int) / 30
-
+    df_true_pos["time_from_pred_to_event"] = df_true_pos[
+        "time_from_pred_to_event"
+    ] / np.timedelta64(
+        1,
+        bin_unit,
+    )
```
Comment (MartinBernstorff) [inline src/psycop_model_evaluation/binary/time/timedelta_plots.py:206]: Probably want to just use `# type: ignore` for this. Unfortunately, dividing a series with `numpy` doesn't play nice. The reason being that mypy can't know what types are in `df_true_pos[
        "time_from_pred_to_event"
    ]`, and therefore can't know whether they are divisible by timedeltas.

File: src/psycop_model_evaluation/binary/time/timedelta_plots.py [hunk 4]
```diff
@@ -216,10 +220,18 @@ def plot_time_from_first_positive_to_event(
     x_labels = list(counts["time_from_first_positive_to_event_binned"])
     y_values = counts[0].to_list()
 
+    bin_unit2str = {
+        "h": "Hours",
+        "D": "Days",
+        "M": "Months",
+        "Q": "Quarters",
+        "Y": "Years",
+    }
```
Comment (MartinBernstorff) [inline src/psycop_model_evaluation/binary/time/timedelta_plots.py:229]:
```suggestion
    bin_unit2str = {
        "h": "Hours",
        "D": "Days",
        "M": "Months",
        "Q": "Quarters",
        "Y": "Years",
    }
```

Excellent! We'll probably need these in other functions as well, so makes sense to extract them out of the function.

**Review by MartinBernstorff — CHANGES_REQUESTED** [review]
Looks good! Minor changes.

**Comment by MartinBernstorff** [comment]
@sarakolding, do you want this functionality? :-)

**Review by MartinBernstorff — APPROVED** [review]

**Pull request merged**